src/path/dec_int.rs

```diff
@@ -55,6 +55,17 @@ impl DecInt {
     }
 }
 
+impl core::fmt::Write for DecInt {
+    fn write_str(&mut self, s: &str) -> core::fmt::Result {
```

Comment:
Could you add `#[inline]` here so that this code can fully inline if called from another crate?